lib/msal-common/src/request/BaseAuthRequest.ts

```diff
@@ -3,20 +3,24 @@
  * Licensed under the MIT License.
  */
 
+import { AuthenticationScheme } from "../utils/Constants";
+
 /**
  * BaseAuthRequest
+ * - authority               - URL of the authority, the security token service (STS) from which MSAL will acquire tokens. Defaults to https://login.microsoftonline.com/common. If using the same authority for all request, authority should set on client application object and not request, to avoid resolving authority endpoints multiple times.
+ * - correlationId           - Unique GUID set per request to trace a request end-to-end for telemetry purposes.
  * - scopes                  - Array of scopes the application is requesting access to.
+ * - authenticationScheme       - The type of token retrieved. Defaults to "Bearer". Can also be type "pop".
  * - claims                  - A stringified claims request which will be added to all /authorize and /token calls
  * - shrClaims               - A stringified claims object which will be added to a Signed HTTP Request
- * - authority               - URL of the authority, the security token service (STS) from which MSAL will acquire tokens. Defaults to https://login.microsoftonline.com/common. If using the same authority for all request, authority should set on client application object and not request, to avoid resolving authority endpoints multiple times.
- * - correlationId           - Unique GUID set per request to trace a request end-to-end for telemetry purposes.
  * - resourceRequestMethod      - HTTP Request type used to request data from the resource (i.e. "GET", "POST", etc.).  Used for proof-of-possession flows.
  * - resourceRequestUri         - URI that token will be used for. Used for proof-of-possession flows.
  */
 export type BaseAuthRequest = {
     authority: string;
     correlationId: string;
     scopes: Array<string>;
+    authenticationScheme?: AuthenticationScheme;
```

Comment:
What are the scenarios where this is not defined? It's always gotta be Bearer or PoP right?

Reply:
Is this just so this isn't a breaking change in common?

Reply:
There's logic that sets it to bearer by default if it isn't set explicitly in the request, so adding it to the initial object is optional. I I think making it optional is hiding implementation details from clients that don't require POP.

Reply:
The platform libs have their own request objects that choose which parameters to expose to the end user. Making this required here shouldn't affect whether or not it's optional for users but helps the type checker enforce that it's defined internally

Reply:
That said, making it required here may be breaking for types that didn't already have it so it can wait until the next common major version bump.